core/common/src/Code.ts

```diff
@@ -34,16 +34,16 @@ export class Code implements CodeProps {
   /** the [CodeValue]($docs/bis/intro/codes.md#codevalue-property) of the Element */
   public value?: string;
 
-  constructor(val: CodeProps) {
-    this.spec = Id64.fromJSON(val.spec);
-    this.scope = JsonUtils.asString(val.scope, "");
-    this.value = JsonUtils.asString(val.value);
+  constructor(codeProps: CodeProps) {
+    this.spec = Id64.fromJSON(codeProps.spec);
```

Comment:
Should make the public `value` a property and trim input in setter?

Reply:
Done

Reply:
Why the question mark after val in the setter?

Reply:
It seemed better to have `undefined` clear rather than throw an error.